cmd/export_test.go

```diff
@@ -0,0 +1,487 @@
+
+	password := "toto"
+
+	viper.Reset()
+	viper.Set(configKeyPassword, password)
+	viper.Set(configKeyStorage, "mock")
+
+	s := store.NewStore()
+	outputFile := ".env"
+
+	testVal := []byte("world")
+	err := s.Set("hello", testVal)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testVal = []byte("world2")
+	err = s.Set("hello2", testVal)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	data, err := store.WriteStore([]byte(password), s)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mockBackend.EXPECT().ExistsContext(ctxMatcher).Return(true, nil)
+	mockBackend.EXPECT().LoadContext(ctxMatcher).Return(data, nil)
+
+	exportCmd.Flags().Set("output", outputFile)
+	exportCmd.Flags().Set("format", "dotenv")
+
+	err = exportCmd.RunE(exportCmd, []string{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// check that the file was created
+	_, err = os.Stat(outputFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// check that the file contains the expected content
+	content, err := ioutil.ReadFile(outputFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected := "hello=\"world\"\nhello2=\"world2\""
+
+	err = os.Remove(outputFile)
+
+	if !isSimilar(string(content), expected) {
+		if err != nil {
+			t.Fatalf("expected '%s', got '%s'. Also, failed to delete out file '%s'", expected, string(content), outputFile)
+		}
+		t.Fatalf("expected '%s', got '%s'", expected, string(content))
+	}
+
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestExportCmdJsonFile(t *testing.T) {
+	hijack()
+	defer restore()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockBackend := NewMockBackend(ctrl)
+	backend.Backends["mock"] = newMockFactory(mockBackend)
+
+	password := "toto"
+
+	viper.Reset()
+	viper.Set(configKeyPassword, password)
+	viper.Set(configKeyStorage, "mock")
+
+	s := store.NewStore()
+	outputFile := "test.json"
+
+	testVal := []byte("world")
+	err := s.Set("hello", testVal)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testVal = []byte("world2")
+	err = s.Set("hello2", testVal)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	data, err := store.WriteStore([]byte(password), s)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mockBackend.EXPECT().ExistsContext(ctxMatcher).Return(true, nil)
+	mockBackend.EXPECT().LoadContext(ctxMatcher).Return(data, nil)
+
+	exportCmd.Flags().Set("output", outputFile)
+	exportCmd.Flags().Set("format", "json")
+
+	err = exportCmd.RunE(exportCmd, []string{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// check that the file was created
+	_, err = os.Stat(outputFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// check that the file contains the expected content
+	content, err := ioutil.ReadFile(outputFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected := "{\"hello\":\"world\",\"hello2\":\"world2\"}"
+
+	err = os.Remove(outputFile)
+
+	if !isSimilar(string(content), expected) {
+		if err != nil {
+			t.Fatalf("expected '%s', got '%s'. Also, failed to delete out file '%s'", expected, string(content), outputFile)
+		}
```

Comment:
No need to test the error from `Remove` error in the `isSimilar` case. Keep both concerns separated, for clearer test results.